src/services/io/podio/CMakeLists.txt

```
@@ -33,12 +33,15 @@ get_target_property(EDM4EIC_INTERFACE_INCLUDE_DIRECTORIES EDM4EIC::edm4eic INTER
 
 # Generate the datamodel header files.
 add_custom_command(
-        OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/datamodel_glue.h ${CMAKE_CURRENT_BINARY_DIR}/datamodel_includes.h
-        COMMAND python3 ${PROJECT_SOURCE_DIR}/make_datamodel_glue.py
-                WORKING_DIR=${CMAKE_CURRENT_BINARY_DIR}
-                EDM4HEP_INCLUDE_DIR=${EDM4HEP_INTERFACE_INCLUDE_DIRECTORIES}
-                EDM4EIC_INCLUDE_DIR=${EDM4EIC_INTERFACE_INCLUDE_DIRECTORIES}
-        MAIN_DEPENDENCY ${PROJECT_SOURCE_DIR}/make_datamodel_glue.py
+    OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/datamodel_glue.h ${CMAKE_CURRENT_BINARY_DIR}/datamodel_includes.h
+    COMMAND python3 ${PROJECT_SOURCE_DIR}/make_datamodel_glue.py
+        WORKING_DIR=${CMAKE_CURRENT_BINARY_DIR}
+        EDM4HEP_INCLUDE_DIR=${EDM4HEP_INTERFACE_INCLUDE_DIRECTORIES}
+        EDM4EIC_INCLUDE_DIR=${EDM4EIC_INTERFACE_INCLUDE_DIRECTORIES}
+    MAIN_DEPENDENCY ${PROJECT_SOURCE_DIR}/make_datamodel_glue.py
+    DEPENDS
+        ${EDM4HEP_INTERFACE_INCLUDE_DIRECTORIES}/edm4hep/EDM4hepVersion.h
+        ${EDM4EIC_INTERFACE_INCLUDE_DIRECTORIES}/edm4eic/EDM4eicVersion.h
```

We can get some forward compat with something like
```suggestion
        ${EDM_VERSION_INCLUDES}
```
and
```cmake
set(EDM_VERSION_INCLUDES )
if(${EDM4HEP_VERSION} VERSION_GREATER_EQUAL "0.9")
list(APPEND EDM_VERSION_INCLUDES ${EDM4HEP_INTERFACE_INCLUDE_DIRECTORIES}/edm4hep/EDM4hepVersion.h)
endif()
if(${EDM4EIC_VERSION} VERSION_GREATER_EQUAL "2.1.0")
list(APPEND EDM_VERSION_INCLUDES ${EDM4EIC_INTERFACE_INCLUDE_DIRECTORIES}/edm4eic/EDM4eicVersion.h)
endif()
```

---

or does cmake complain about that?

---

Note commented APPENDs

---

No issues with EDM4hep < 0.9 and EDM4eic < 2.1? That is the corner case that leads to an empty DEPENDS, right.

---

The logic of my test is that, if you comment out appends, it executes as if includes are not available.

---

Right, I see the comment characters now. Narrow mobile interface not always the best...